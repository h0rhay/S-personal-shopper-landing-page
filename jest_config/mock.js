export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
