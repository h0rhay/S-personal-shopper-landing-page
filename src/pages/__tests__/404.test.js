import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../jest_config/mock';
import ErrorPage from '../404';

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <ErrorPage />
    </ThemeProvider>
  );

describe('404 - Error page', () => {
  it('should match screenshot', () => {
    const { container } = renderComponent({ theme: mockTheme });
    expect(container).toMatchSnapshot();
  });
});
