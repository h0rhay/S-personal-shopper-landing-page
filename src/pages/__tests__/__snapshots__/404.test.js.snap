// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 - Error page should match screenshot 1`] = `
<div>
  <div
    class="sc-bdvvtL kicZDN"
  >
    <a
      class="sc-gsDKAQ"
      href="https://www.selfridges.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <test-file-stub
        classname="sc-dkPtRN fFrQVd"
      />
    </a>
    <h1
      class="sc-hKwDye gDhSfz"
    >
      404
    </h1>
    <p
      class="sc-eCImPb EPUDe"
    >
      Sorry, the page you requested does not exist.
    </p>
  </div>
</div>
`;
